### ISSUE TITLE
Add "open in editor" action

### ISSUE THREAD

**Issue by fkling** [sourcegraph/sourcegraph#61629]
```[tasklist]
### Tasks
- [x] Implement new button
- [x] Add automated tests
- [ ] Implement popover to configure editor
```

**Comment by bahrmichael** [comment]
Recap of what the OpenInEditorActionItem.tsx is currently doing (i.e. the things we need to do in the svelte version):

- Retrieving settings from the backend, and error handling (request fails or settings are invalid)
- Popover to change the editor integration (and its related race conditions)
- Handling multiple editors (e.g. Show a button with the editor icon for each registered editor )
- Constructing the URL that opens the file in the IDE
- Not showing the dropdown on small viewports, since we assume they won't have an IDE on their phones anyway

Open questions:
- Can we drop the code to migrate legacy settings? It's from September '22.

**Comment by bahrmichael** [comment]
Here's where the button would be added (once that PR is merged): https://github.com/sourcegraph/sourcegraph/pull/61723/files#diff-21381efc7ce3a33522c1ffcc88edefa0ff8273c5a9b9d2c7447bd15adc4a6736R104-R109

**Comment by camdencheek** [comment]
@bahrmichael can this be closed after #61782?

**Comment by bahrmichael** [comment]
Will move the remaining work to a new task 👍
